data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/s3/S3SinkConfig.java

```diff
@@ -30,8 +32,8 @@ public class S3SinkConfig {
     private AwsAuthenticationOptions awsAuthenticationOptions;
 
     @JsonProperty("bucket")
-    @NotNull
     @NotEmpty
+    @Size(min = 3, max = 63, message = "bucket lengthy should be between 3 and 63 characters")
```

Comment:
While "bucket lengthy" is kind of cute, we probably should correct that. :)

Also, I'm not quite sure that having this restriction is right. It binds our code to the _current_ S3 restrictions. They may or may not change.

Reply:
Okay, I can remove it Just tried keeping in consistent with dlq config. With prefix support, the length can be more than 63.

Reply:
> With prefix support, the length can be more than 63.

That is a good point as well.

I think we can remove it. Maybe we keep the minimum to 3. I'm guessing they have a strong reason to limit that. But, I could imagine S3 might extend the maximum size.

We can still have some sort of sanity check value (say 1000) to ensure that users aren't creating bucket names that are tons of YAML, code, or other garbage inputs.